agent/mibgroup/if-mib/data_access/interface.c

```diff
@@ -908,6 +940,31 @@ _free_interface_config(void)
     conf_list = NULL;
 }
 
+/*
+ * Maximum number of interfaces to include in IF-MIB
+ */
+static void
+_parse_ifmib_max_num_ifaces(const char *token, char *cptr)
+{
+    long long_val;
+    char *name, *st;
+
+    errno = 0;
+    name = strtok_r(cptr, " \t", &st);
+    if (!name) {
+        config_perror("Missing NUMBER parameter");
+        return;
+    }
+    long_val = strtol(cptr, NULL, 10);
+    if (errno) {
+           config_perror("Error converting parameter");
+           return;
+    }
```

Comment:
From [https://pubs.opengroup.org/onlinepubs/009695399/functions/strtol.html](https://pubs.opengroup.org/onlinepubs/009695399/functions/strtol.html):

>     Upon successful completion, these functions shall return the converted value, if any. If no conversion
>     could be performed, 0 shall be returned [CX] [Option Start]  and errno may be set to [EINVAL].
> 
>     If the correct value is outside the range of representable values, {LONG_MIN}, {LONG_MAX}, {LLONG_MIN},
>     or {LLONG_MAX} shall be returned (according to the sign of the value), and errno set to [ERANGE].
> 

I think this means that it is not guaranteed that errno is set if strtol() fails. How about using sscanf() instead?

Reply:
Yes, thanks.  I changed this to an sscanf in my update.